### PR TITLE
Add samples conventions

### PR DESCRIPTION
**Pull request by bmuschko**
Provides appropriate code and output directories as attributes to Asciidoctor. The expected directory structure in a guides project looks as follows. Samples code and expected output can be defined in distinct directories.

```
.
└── samples
    ├── code
    └── output
```

Defines samples directory to `test` task as system property for further processing. For more info see the following [code](https://github.com/gradle-guides/designing-gradle-plugins/blob/master/build.gradle#L11-L25) and [code review comment](https://github.com/gradle-guides/designing-gradle-plugins/pull/5#pullrequestreview-48527528). This now has been rolled into the plugin.

To follow: A base class extending Spock's Specification to reduce boiler code when testing sample code.

Uses the regular Apache 2 license across all files. No need to use an external plugin to enforce it. As needed FindBugs can be used to enforce the license header.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -40,26 +39,11 @@ dependencies {
     runtime 'org.ysb33r.gradle:gradle-cloudci-plugin:1.0'
     runtime 'org.ysb33r.gradle:gradletest:1.0-beta11'
 
-    testImplementation "org.spockframework:spock-core:1.0-groovy-${spockGroovyVer}", {
+    testCompile "org.spockframework:spock-core:1.0-groovy-${spockGroovyVer}", {
```
Comment (ysb33r) [inline build.gradle]: Out of curiosity why did you revert back to the legacy configuration?

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,14 @@
 plugins {
```
Comment (ysb33r) [inline build.gradle]: Remember to update the plugin version number. I guess `0.7` should do as this is the start of a departure from the original plugin.

Comment (bmuschko) [inline build.gradle]: Didn't make it part of this change as I am not sure if there we were other changes waiting and potentially conflicting. We'll definitely change the version number before releasing.

File: build.gradle (hunk 1 above)
Comment (bmuschko) [inline build.gradle]: My bet. I forgot about it when I made some change to the functional test source set. Fix both places to use the new configurations.

File: build.gradle (hunk 2 above)
Comment (ysb33r) [inline build.gradle]: I'm keeping it as is for now and not merging.

Comment (ysb33r) [inline build.gradle]: Will merge it and release an update by start of next week, if no more changes arrived in the mean time.

**Comment by bmuschko** [comment]
> I think the license plugin is a cool idea actually, at least if it does what I think (auto-adding license headers).

The description of the plugin is somewhat unclear on it: "This plugin will scan and adapt your source files to include a provided header, e.g. a LICENSE file.". When I tried it, it only failed my build.

For this change I wanted to make sure that we have a person-agnostic copyright in place. We can think of other options down the road though I don't expect this project to grow massively over time.

**Comment by ysb33r** [comment]
> The description of the plugin is somewhat unclear on it: "This plugin will scan and adapt your source files to include a provided header, e.g. a LICENSE file.". When I tried it, it only failed my build.

Probably because you left a header off a file OR your license temlpate did not match the current headers.

**Comment by ysb33r** [comment]
I'll do a release today, @bmuschko

**Pull request merged**